ITL-2 Implement Todo list

BiroAron opened the pull request.

BiroAron committed:
ITL-7 Create the header component

BiroAron committed:
ITL-8 Add the title

BiroAron committed:
ITL-9 Add an add button, include a icon library or include the icon from the design

BiroAron committed:
ITL-10 Create the event that triggers the todo creation functionality

BiroAron committed:
Replaced plain CSS with Tailwind

BiroAron committed:
Made all the changes regarding to the comments

BiroAron committed:
Made the changes regarding to class names and reactive variables

BiroAron committed:
Removed all unnecessary code and comments

BiroAron committed:
Removed unnecessary id-s and redundant script

BiroAron committed:
Merge branch 'feature/ITL-1-implement-page-header' into develop

BiroAron committed:
ITL 11 Create the Placeholder img when the user is in the home page and no todos added to todo list

BiroAron committed:
ITL 12 Mobile friendly looking

BiroAron committed:
ITL 13  make the todo list look

gerebdavid commented on src/components/Header.vue:
```diff
@@ -0,0 +1,26 @@
+<script setup lang="ts">
+import PlusIcon from './PlusIcon.vue'
+
+const emit = defineEmits<{ (e: 'toggleVisibility'): void }>()
```
You dont have to type emits if they dibt have parameters. defineEmits(["toggleVisibility"]) will do the job

gerebdavid commented on src/components/ToDoForm.vue:
```diff
@@ -1,42 +1,82 @@
     </div>
-
+    <div>
+      <div>
```
Do you need this div?

gerebdavid commented on src/components/ToDoForm.vue:
```diff
@@ -1,42 +1,82 @@
+  title: '',
+  priority: 'Low',
+  text: '',
+  state: false
```
Whats this state key for?

gerebdavid commented on src/components/ToDoForm.vue:
```diff
@@ -1,42 +1,82 @@
-         }
-     }
+function addItem() {
+  if (newTodo.title !== '') {
```
Its enough if you check if newTodo.title has value, like if (newTodo.title) {

gerebdavid commented on src/components/ToDoItems.vue:
```diff
@@ -0,0 +1,85 @@
+          <div class="flex justify-center items-end phone:items-center">
+            <div
+              class="w-8 h-8 rounded-full border-4 border-black relative"
+              :class="todo.state ? 'border-green-500' : 'border-black'"
```
create a computed from this

gerebdavid commented on src/components/ToDoItems.vue:
```diff
@@ -0,0 +1,85 @@
+            >
+              <CheckedIcon
+                class="absolute bottom-0 left-0"
+                :class="todo.state ? 'block' : 'hidden'"
```
create a computed from this

gerebdavid commented on src/components/ToDoItems.vue:
```diff
@@ -0,0 +1,85 @@
+defineProps(['todos'])
+const emit = defineEmits<{ (e: 'deleteItem', index: number): void }>()
+
+function updatePriorityClass(priority: string) {
```
Create a colorMap variable and use that instead of this function

gerebdavid commented on src/components/ToDoList.vue:
```diff
@@ -1,45 +1,68 @@
+      <Header @toggle-visibility="toggleVisibility"></Header>
+      <div class="flex justify-center align-center">
+        <div
+          :class="isNewElementFormActive ? 'block' : 'hidden'"
```
create a computed from this

gerebdavid commented on src/components/ToDoList.vue:
```diff
@@ -1,45 +1,68 @@
+      <ToDoItem :todos="todos" @delete-item="removeItem" />
+      <div class="flex justify-center align-middle"></div>
+      <div
+        :class="shouldEmptyListImageBeDisplayed ? 'block' : 'hidden'"
```
create a computed from this

gerebdavid commented on src/components/ToDoList.vue:
```diff
@@ -1,45 +1,68 @@
-		todos.value.push(todo);
-	}
+function addTodo(todo: Todo) {
+  const todoCopy = { ...todo }
```
If you want to clone an object always use the built in structuredClone function, https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

BiroAron committed:
Made changes regarding the comments

-removed all of the conditional operators from the template tags
-removed unnecessary complicated code
-renamed non-descriptive variable names
-created a colorMap

BiroAron committed:
updatePriorityClass is now a variable instead of a function
